pkg/ingester/ingester.go

```diff
@@ -97,36 +99,42 @@ func New(phlarectx context.Context, cfg Config, dbConfig phlaredb.Config, storag
 		return nil, err
 	}
 
-	i.lifecyclerWatcher = services.NewFailureWatcher()
-	i.lifecyclerWatcher.WatchService(i.lifecycler)
+	rpEnforcer := newRetentionPolicyEnforcer(i, defaultRetentionPolicy())
+	i.subservices, err = services.NewManager(i.lifecycler, rpEnforcer)
+	if err != nil {
+		return nil, errors.Wrap(err, "services manager")
+	}
+	i.subservicesWatcher = services.NewFailureWatcher()
+	i.subservicesWatcher.WatchManager(i.subservices)
 	i.Service = services.NewBasicService(i.starting, i.running, i.stopping)
 	return i, nil
 }
 
 func (i *Ingester) starting(ctx context.Context) error {
-	// pass new context to lifecycler, so that it doesn't stop automatically when Ingester's service context is done
-	err := i.lifecycler.StartAsync(context.Background())
-	if err != nil {
-		return err
-	}
-
-	err = i.lifecycler.AwaitRunning(ctx)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return services.StartManagerAndAwaitHealthy(ctx, i.subservices)
 }
```

Comment:
I failed to find why we need to use the background context, and the server seems to be working fine without it. But I'm not 100% sure if there's a catch